finish/src/main/java/io/openliberty/guides/system/SystemResource.java

```diff
@@ -12,36 +12,22 @@
  // end::copyright[]
 package io.openliberty.guides.system;
 
+// CDI
+import javax.enterprise.context.RequestScoped;
+import javax.ws.rs.GET;
 // JAX-RS
 import javax.ws.rs.Path;
-import javax.ws.rs.core.MediaType;
-import javax.ws.rs.GET;
 import javax.ws.rs.Produces;
-
-// JSON-P
-import javax.json.Json;
-import javax.json.JsonObject;
-import javax.json.JsonObjectBuilder;
-
-// CDI
-import javax.enterprise.context.RequestScoped;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
 
 @RequestScoped
 @Path("properties")
-public class PropertiesResource {
-
-    @GET
-    @Produces(MediaType.APPLICATION_JSON)
-    public JsonObject getProperties() {
-        JsonObjectBuilder builder = Json.createObjectBuilder();
-
-        System.getProperties()
-              .entrySet()
-              .stream()
-              .forEach(entry -> builder.add((String)entry.getKey(),
-                                            (String)entry.getValue()));
+public class SystemResource {
 
-        return builder.build();
-    }
-    
+  @GET
+  @Produces(MediaType.APPLICATION_JSON)
+  public Response getProperties() {
+    return Response.ok(System.getProperties()).build();
+  } 
```

Comment:
Actually, I think the Properties can be returned directly. But I want to get Artur's opinion on it. Thanks.

Reply:
Yes, it can be returned directly. Response is useful when there custom/multiple HTTP codes or models returned.  Also when returning a POJO, if method returns null then status is set to 204(No Content)

Reply:
Thank you for your review and reply.